pkg/flatbuffers/Kconfig

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2022 Inria
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config PACKAGE_FLATBUFFERS
+    bool "Flatbuffers package"
+    depends on TEST_KCONFIG
+
+    select MODULE_CPP11-COMPAT
```

Comment:
All these modules that are bringing cpp11-compat should depend on `HAS_CPP` and `HAS_LIBSTDCPP`, as you are using `select` instead of `depends on MODULE_CPP11-COMPAT`

Reply:
BTW this has to do with a trade-off between using `select` vs. `depends on`. I started a conversation regarding this in https://github.com/RIOT-OS/RIOT/issues/15429. Some good documentation on the matter can be found in https://docs.zephyrproject.org/latest/build/kconfig/tips.html#select-statements

Reply:
Isn't this automatic since cpp11-compat already depends on HAS_CPP and HAS_LIBSTDCPP ?

https://github.com/RIOT-OS/RIOT/blob/cdf1d43e1551558f09284f05a4fdb88f47ea6606/sys/cpp11-compat/Kconfig#L11-L12

Reply:
No, because `select` defines a [reverse dependency](https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html#menu-attributes), which will try to force the value, regardless of the dependencies of the selected symbol (although it would trigger an error in the final check, I find it misleading to show the option when in reality one can't choose it). `depends on` on the other hand will respect the dependencies of the other symbol, but makes configurations more verbose.

Reply:
`In general use select only for non-visible symbols (no prompts anywhere) and for symbols with no dependencies`. It seems to me that is not followed most of the time.

Reply:
> `In general use select only for non-visible symbols (no prompts anywhere) and for symbols with no dependencies`. It seems to me that is not followed most of the time.

Indeed. The initial idea was to follow that guideline, but people in RIOT are used to a more automagic way of bringing modules, so during the migration we used `select` more and more. It has some side effects like having to move dependencies up in the chain, otherwise you may end up selecting a symbol that does not meet its dependencies, but allows for a shorter configuration. Again, it is a trade off. Personally I'd try to only make use of `depends on`, but I can see how that may annoy some users.